packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx

```diff
@@ -31,15 +34,41 @@ export const RainbowKitCustomConnectButton = () => {
 
               if (chain.unsupported || chain.id !== configuredNetwork.id) {
                 return (
-                  <>
-                    <span className="text-xs" style={{ color: networkColor }}>
-                      {configuredNetwork.name}
-                    </span>
-                    <button className="btn btn-sm btn-error ml-2" onClick={openChainModal} type="button">
+                  <div className="dropdown dropdown-end">
+                    <button tabIndex={0} className="btn btn-error btn-sm dropdown-toggle">
                       <span>Wrong network</span>
                       <ChevronDownIcon className="h-6 w-4 ml-2 sm:ml-0" />
                     </button>
-                  </>
+                    <ul tabIndex={0} className="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-56">
+                      <li>
+                        <button
+                          className="menu-item"
+                          type="button"
+                          onClick={() => {
+                            if (switchNetwork) {
+                              switchNetwork(configuredNetwork.id);
+                            }
+                          }}
+                        >
+                          <ArrowsRightLeftIcon className="h-6 w-4 ml-2 sm:ml-0" />
+                          <span>
```

Comment:
Add `className="whitespace-nowrap"` and then on line 42, remove `w-56` to to let the dropdown set its width based on the text (allows it to vary based on what the target network name is)

Reply:
Thanks! Cool to know about `whitespace-nowrap` utility